### ISSUE TITLE
Failed to build variable mapping expression

### ISSUE THREAD

**Issue by saig0** [camunda-cloud/zeebe#7132]
**Describe the bug**
If I deploy a process that contains the following input mapping then the deployment is rejected with the following message.

Input mapping:
```
msisdn
= (msisdnList [1])
```

Failure message:
```
Failed to build variable mapping expression: failed to parse expression '{msisdn: (msisdnList [1])}': Expected (filterContinuation | pathContinuation | end-of-input):1:26, found "}"
```

Additionally, the following failure is logged in the error reporting tool:

```
java.lang.IllegalStateException: Failed to build variable mapping expression: failed to parse expression '{msisdn: (msisdnList [1])}': Expected (filterContinuation | pathContinuation | end-of-input):1:26, found "}"
``` 

**To Reproduce**
1. deploy a process with a service task that has the following input mapping
```
msisdn
= (msisdnList [1])
```
2. verify that the deployment is rejected

**Expected behavior**
1. The process is deployed successfully. 
2. The failure is handled like other violations. (Currently, the violation is not covered by the validation phase. Instead, an exception is thrown on transforming the process.)

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.IllegalStateException: Failed to build variable mapping expression: failed to parse expression '{msisdn: (msisdnList [1])}': Expected (filterContinuation | pathContinuation | end-of-input):1:26, found "}"
at io.camunda.zeebe.engine.processing.deployment.model.transformer.VariableMappingTransformer.parseExpression (VariableMappingTransformer.java:187)
at io.camunda.zeebe.engine.processing.deployment.model.transformer.VariableMappingTransformer.transformInputMappings (VariableMappingTransformer.java:83)
at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.lambda$transformIoMappings$1 (FlowNodeTransformer.java:64)
at java.util.Optional.map (Unknown Source)
at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transformIoMappings (FlowNodeTransformer.java:62)
at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transform (FlowNodeTransformer.java:38)
at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transform (FlowNodeTransformer.java:21)
at io.camunda.zeebe.engine.processing.deployment.model.transformation.TransformationVisitor.visit (TransformationVisitor.java:42)
at io.camunda.zeebe.model.bpmn.traversal.TypeHierarchyVisitor.visit (TypeHierarchyVisitor.java:39)
at io.camunda.zeebe.model.bpmn.traversal.ModelWalker.walk (ModelWalker.java:61)
at io.camunda.zeebe.engine.processing.deployment.model.transformation.BpmnTransformer.transformDefinitions (BpmnTransformer.java:102)
at io.camunda.zeebe.engine.processing.deployment.transform.DeploymentTransformer.transformResource (DeploymentTransformer.java:123)
at io.camunda.zeebe.engine.processing.deployment.transform.DeploymentTransformer.transform (DeploymentTransformer.java:96)
at io.camunda.zeebe.engine.processing.deployment.DeploymentCreateProcessor.processRecord (DeploymentCreateProcessor.java:100)
at io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.processRecord (TypedRecordProcessor.java:54)
at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.lambda$processInTransaction$3 (ProcessingStateMachine.java:296)
at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run (ZeebeTransaction.java:84)
at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processInTransaction (ProcessingStateMachine.java:286)
at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processEvent (ProcessingStateMachine.java:254)
at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextEvent (ProcessingStateMachine.java:219)
at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextEvent (ProcessingStateMachine.java:210)
at io.camunda.zeebe.util.sched.ActorJob.invoke (ActorJob.java:76)
at io.camunda.zeebe.util.sched.ActorJob.execute (ActorJob.java:39)
at io.camunda.zeebe.util.sched.ActorTask.execute (ActorTask.java:122)
at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask (ActorThread.java:94)
at io.camunda.zeebe.util.sched.ActorThread.doWork (ActorThread.java:78)
at io.camunda.zeebe.util.sched.ActorThread.run (ActorThread.java:191)
```

</p>
</details>

[Error reporting tool](https://console.cloud.google.com/errors/CKGCu_C7vNvDKA?service=zeebe&time=P7D&refresh=off&project=camunda-cloud-240911)
[Log explorer](https://console.cloud.google.com/logs/query;query=error_group%2528%22CKGCu_C7vNvDKA%22%2529%0AlogName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22production-worker-1%22%0Aresource.labels.namespace_name%3D%2297ae6962-4dc1-4f59-8b75-15ae402f2531-zeebe%22%0Aresource.labels.container_name%3D%22zeebe%22;timeRange=2021-05-26T07:13:09.827Z%2F2021-05-26T08:13:09.827Z;pinnedLogId=2021-05-26T07:42:39.827032Z%2F1j881j4g4ny0jxn;cursorTimestamp=2021-05-26T07:42:39.827032Z?project=camunda-cloud-240911)

**Environment:**
 - OS: Camunda Cloud
 - Zeebe Version: `1.0.0`
 - Configuration:

**Comment by saig0** [comment]
**Workaround**
In this case, the issue can be fixed by changing the input mapping to
```
msisdn
= msisdnList [1]
```

(no parenthesis)

**Comment by saig0** [comment]
I created an issue in the FEEL engine to parse the expression: camunda/feel-scala#278

**Comment by saig0** [comment]
The issue should be fixed by #7682. Let's verify it manually.

**Comment by saig0** [comment]
I tested the behavior manually with the latest `1.2.0-SHAPSHOT`. I can deploy the process and create an instance. The input mapping is applied and extracts the first element of the list. :heavy_check_mark:
